alchemist-mix.el

```diff
@@ -154,22 +167,29 @@
   (interactive "Mmix local.install URL: ")
   (alchemist-mix-execute (list "local.install" url)))
 
-(defun alchemist-mix-hex-search (command)
-  "Display packages matching the given search query."
-  (interactive "Mmix hex.search: ")
-  (alchemist-mix-execute (list "hex.search" command)))
-
-(defun alchemist-mix-help (command)
-  "Show help output for a specific mix command."
-  (interactive "Mmix help: ")
-  (alchemist-mix-execute (list "help" command)))
-
-(defun alchemist-mix-execute (cmdlist)
-  "Run a mix command."
-  (interactive "Mmix: ")
-  (let ((old-directory default-directory))
+(defun alchemist-mix-hex-search (command &optional prefix)
+  "Display packages matching the given search query. Prompt for the mix env
+if the prefix arg is set."
+  (interactive "Mmix hex.search: \nP")
+  (alchemist-mix-execute (list "hex.search" command) prefix))
+
+(defun alchemist-mix-help (command &optional prefix)
+  "Show help output for a specific mix command. Prompt for the mix env if
+the prefix arg is set."
+  (interactive "Mmix help: \nP")
```

Comment:
Checking the info pages on `interactive`:

```
There are three possibilities for the argument ARG-DESCRIPTOR:
  ...
   * It may be a string; its contents are a sequence of elements
     separated by newlines, one for each argument(1).  Each element
     consists of a code character (*note Interactive Codes::)
     optionally followed by a prompt (which some code characters use
     and some ignore).  Here is an example:

          (interactive "P\nbFrobnicate buffer: ")
```

So, AFAIK the character codes make sense with the `\n`.